pkg/apis/contrail/v1alpha1/swiftproxy_types.go

```diff
@@ -68,17 +68,17 @@ type SwiftProxyList struct {
 // SwiftProxyInstanceType is type unique name used for labels
 const SwiftProxyInstanceType = "SwiftProxy"
 
-// PodsCertSubjects gets list of SwiftProxy pods certificate subjets which can be passed to the certificate API
-func (s *SwiftProxy) PodsCertSubjects(podList *corev1.PodList, serviceIP string) []certificates.CertificateSubject {
-	altIPs := PodAlternativeIPs{ServiceIP: serviceIP}
+//PodsCertSubjects gets list of SwiftProxy pods certificate subjets which can be passed to the certificate API
+func (s *SwiftProxy) PodsCertSubjects(podList *corev1.PodList, serviceIP string, alternativeIP ...string) []certificates.CertificateSubject {
+	altIPs := PodAlternativeIPs{ServiceIP: serviceIP, Retriever: func(pod corev1.Pod) []string { return alternativeIP }}
 	return PodsCertSubjects(podList, s.Spec.CommonConfiguration.HostNetwork, altIPs)
 }
 
-// GetServiceType returns chosen Service type for Swift Proxy, default is LoadBalancer
+// GetServiceType returns chosen Service type for Swift Proxy, default is ClusterIP
 func (s *SwiftProxy) GetServiceType() corev1.ServiceType {
 	st := s.Spec.ServiceConfiguration.Service.GetServiceType()
 	if st == "" {
-		return corev1.ServiceTypeLoadBalancer
+		return corev1.ServiceTypeClusterIP
```

Comment:
LoadBalancer service should be the default value. The default ServiceTypeClusterIP will break other projects

Reply:
Corrected.